### PR TITLE
remove graph-repository-connector dependency from connector-configuration-factory

### PR DESCRIPTION
**Pull request by bogdan-sava**
# Description

remove graph-repository-connector dependency from connector-configuration-factory

### CODE REVIEW

**Review by planetf1 — CHANGES_REQUESTED** [review]
Looks like a great update! I just noticed one very minor issue with an explicit version in the pom.

I'm going to pull & build/check as an extra test

File: open-metadata-distribution/open-metadata-assemblies/src/main/assemblies/egeria-omag.xml [hunk 1]
```diff
@@ -71,6 +71,7 @@
                 <include>org.odpi.egeria:graph-repository-connector:*</include>
                 <include>org.odpi.egeria:inmemory-open-metadata-topic-connector:*</include>
                 <include>org.odpi.egeria:inmemory-repository-connector:*</include>
+                <include>org.odpi.egeria:graph-repository-connector:*</include>
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-distribution/open-metadata-assemblies/src/main/assemblies/egeria-omag.xml]: this is done now

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/graph-repository-connector/pom.xml [hunk 2]
```diff
@@ -165,6 +165,27 @@
                     </execution>
                 </executions>
             </plugin>
+
+            <plugin>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <version>3.3.0</version>
```
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/graph-repository-connector/pom.xml:171]: In fact it's already in the top level pom, so just removing the version line should be fine
(It has no impact for now, it's just reducing number of changes in future)

**Comment by planetf1** [comment]
I'm seeing an error when testing the PR in launching the ui chassis - which is hitting an exception from CassandraHealthContributor . We saw this in egeria previously, and disabled the health contributor in the chassis properties.

 management.health.cassandra.enabled=false
 
 This wasn't done in the UI - but we hadn't seen the issue
 
Not sure why this change is causing this now - some difference in component scan, or a new version

ie 
```
2021-11-09 12:03:25.090 -ERROR 91413 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'cassandraHealthContributor' defined in class path resource [org/springframework/boot/actuate/autoconfigure/cassandra/CassandraHealthContributorConfigurations$CassandraReactiveDriverConfiguration.class]: Unsatisfied dependency expressed through method 'cassandraHealthContributor' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraSession' defined in class path resource [org/springframework/boot/autoconfigure/cassandra/CassandraAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.datastax.oss.driver.api.core.CqlSession]: Factory method 'cassandraSession' threw exception; nested exception is com.datastax.oss.driver.api.core.AllNodesFailedException: Could not reach any contact point, make sure you've provided valid addresses (showing first 1 nodes, use getAllErrors() for more): Node(endPoint=/127.0.0.1:9042, hostId=null, hashCode=1ea091d7): [com.datastax.oss.driver.api.core.connection.ConnectionInitException: [s0|control|connecting...] Protocol initialization request, step 1 (OPTIONS): failed to send request (io.netty.channel.StacklessClosedChannelException)]
```

I have a shell script to setup a local deployment which hasn't changed (though the broader environment could have)

* It could be specific to my environment 
* If the UI works ok for you without cassandra etc configured we could merge into master & then run more tests ie against the charts & fix quickly if needed
* alternatively I could test in more environments/with charts - though this will take longer

File: open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/graph-repository-connector/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (planetf1) [reply open-metadata-implementation/adapters/open-connectors/repository-services-connectors/open-metadata-collection-store-connectors/graph-repository-connector/pom.xml]: thanks for updating

**Comment by planetf1** [comment]
This is the sequence that causes the failure
```
$ export LOADER_PATH=~/src/egeria/open-metadata-distribution/open-metadata-assemblies/target/egeria-3.4-SNAPSHOT-distribution/egeria-omag-3.4-SNAPSHOT/server/lib
jonesn:egeria/ ((0aab3cad73...)*) $ ls $LOADER_PATH                                                                           [15:28:29]
audit-log-console-connector-3.4-SNAPSHOT.jar                      governance-action-connectors-3.4-SNAPSHOT.jar
audit-log-event-topic-connector-3.4-SNAPSHOT.jar                  governance-services-sample-3.4-SNAPSHOT.jar
audit-log-file-connector-3.4-SNAPSHOT.jar                         graph-repository-connector-3.4-SNAPSHOT-jar-with-dependencies.jar
audit-log-slf4j-connector-3.4-SNAPSHOT.jar                        inmemory-open-metadata-topic-connector-3.4-SNAPSHOT.jar
avro-file-connector-3.4-SNAPSHOT.jar                              inmemory-repository-connector-3.4-SNAPSHOT.jar
basic-file-connector-3.4-SNAPSHOT.jar                             kafka-integration-connector-3.4-SNAPSHOT.jar
cohort-registry-file-store-connector-3.4-SNAPSHOT.jar             kafka-open-metadata-topic-connector-3.4-SNAPSHOT.jar
configuration-encrypted-file-store-connector-3.4-SNAPSHOT.jar     omrs-rest-repository-connector-3.4-SNAPSHOT.jar
configuration-file-store-connector-3.4-SNAPSHOT.jar               open-lineage-janus-connector-3.4-SNAPSHOT.jar
csv-file-connector-3.4-SNAPSHOT.jar                               open-metadata-archive-file-connector-3.4-SNAPSHOT.jar
data-folder-connector-3.4-SNAPSHOT.jar                            open-metadata-security-samples-3.4-SNAPSHOT.jar
discovery-service-connectors-3.4-SNAPSHOT.jar                     openapi-integration-connector-3.4-SNAPSHOT.jar
elasticsearch-integration-connector-3.4-SNAPSHOT.jar              openlineage-integration-connectors-3.4-SNAPSHOT.jar
files-integration-connectors-3.4-SNAPSHOT.jar                     spring-rest-client-connector-3.4-SNAPSHOT.jar
jonesn:egeria/ ((0aab3cad73...)*) $ OMAS_SERVER_URL=https://localhost:9444 OMAS_SERVER_NAME=cocoMDS4 java -jar ./open-metadata-distribution/open-metadata-assemblies/target/egeria-3.4-SNAPSHOT-distribution/egeria-omag-3.4-SNAPSHOT/user-interface/ui-chassis-spring-3.4-SNAPSHOT.jar
 ODPi Egeria

       ______                        _                  __  __   ____
      / ____/ ____ _  ___    _____  (_) ____ _         / / / /  /  _/
     / __/   / __ `/ / _ \  / ___/ / / / __ `/        / / / /   / /
    / /___  / /_/ / /  __/ / /    / / / /_/ /        / /_/ /  _/ /
   /_____/  \__, /  \___/ /_/    /_/  \__,_/         \____/  /___/
           /____/


 :: Powered by Spring Boot (v2.5.6) ::

2021-11-09 15:28:48.706 - INFO 99716 --- [           main] o.o.o.u.u.springboot.EgeriaUIPlatform    : Starting EgeriaUIPlatform using Java 11.0.11 on Nigels-MBP.fritz.box with PID 99716 (/Users/jonesn/IdeaProjects/egeria.maven/open-metadata-distribution/open-metadata-assemblies/target/egeria-3.4-SNAPSHOT-distribution/egeria-omag-3.4-SNAPSHOT/user-interface/ui-chassis-spring-3.4-SNAPSHOT.jar started by jonesn in /Users/jonesn/IdeaProjects/egeria.maven)
2021-11-09 15:28:48.710 - INFO 99716 --- [           main] o.o.o.u.u.springboot.EgeriaUIPlatform    : No active profile set, falling back to default profiles: default
2021-11-09 15:28:51.849 - INFO 99716 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8443 (https)
2021-11-09 15:28:58.204 -ERROR 99716 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'cassandraHealthContributor' defined in class path resource [org/springframework/boot/actuate/autoconfigure/cassandra/CassandraHealthContributorConfigurations$CassandraReactiveDriverConfiguration.class]: Unsatisfied dependency expressed through method 'cassandraHealthContributor' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraSession' defined in class path resource [org/springframework/boot/autoconfigure/cassandra/CassandraAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.datastax.oss.driver.api.core.CqlSession]: Factory method 'cassandraSession' threw exception; nested exception is com.datastax.oss.driver.api.core.AllNodesFailedException: Could not reach any contact point, make sure you've provided valid addresses (showing first 1 nodes, use getAllErrors() for more): Node(endPoint=/127.0.0.1:9042, hostId=null, hashCode=21ad6f42): [com.datastax.oss.driver.api.core.connection.ConnectionInitException: [s0|control|connecting...] Protocol initialization request, step 1 (OPTIONS): failed to send request (io.netty.channel.StacklessClosedChannelException)]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800) ~[spring-beans-5.3.12.jar!/:5.3.12]
```

**Comment by planetf1** [comment]
If the spring property is set to disable the cassandra endpoint ie here's the env var equiv - it works:
```
export MANAGEMENT_HEALTH_CASSANDRA_ENABLED=false                                          [15:30:50]
jonesn:egeria/ ((0aab3cad73...)*) $ cd open-metadata-implementation                                                           [15:31:52]
jonesn:egeria/ ((0aab3cad73...)*) $ OMAS_SERVER_URL=https://localhost:9444 OMAS_SERVER_NAME=cocoMDS4 java -jar ./open-metadata-distribution/open-metadata-assemblies/target/egeria-3.4-SNAPSHOT-distribution/egeria-omag-3.4-SNAPSHOT/user-interface/ui-chassis-spring-3.4-SNAPSHOT.jar
 ODPi Egeria

       ______                        _                  __  __   ____
      / ____/ ____ _  ___    _____  (_) ____ _         / / / /  /  _/
     / __/   / __ `/ / _ \  / ___/ / / / __ `/        / / / /   / /
    / /___  / /_/ / /  __/ / /    / / / /_/ /        / /_/ /  _/ /
   /_____/  \__, /  \___/ /_/    /_/  \__,_/         \____/  /___/
           /____/


 :: Powered by Spring Boot (v2.5.6) ::

2021-11-09 15:32:04.176 - INFO 271 --- [           main] o.o.o.u.u.springboot.EgeriaUIPlatform    : Starting EgeriaUIPlatform using Java 11.0.11 on Nigels-MBP.fritz.box with PID 271 (/Users/jonesn/IdeaProjects/egeria.maven/open-metadata-distribution/open-metadata-assemblies/target/egeria-3.4-SNAPSHOT-distribution/egeria-omag-3.4-SNAPSHOT/user-interface/ui-chassis-spring-3.4-SNAPSHOT.jar started by jonesn in /Users/jonesn/IdeaProjects/egeria.maven)
2021-11-09 15:32:04.180 - INFO 271 --- [           main] o.o.o.u.u.springboot.EgeriaUIPlatform    : No active profile set, falling back to default profiles: default
2021-11-09 15:32:07.304 - INFO 271 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8443 (https)
2021-11-09 15:32:11.362 - INFO 271 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8443 (https) with context path ''
2021-11-09 15:32:11.380 - INFO 271 --- [           main] o.o.o.u.u.springboot.EgeriaUIPlatform    : Started EgeriaUIPlatform in 8.177 seconds (JVM running for 9.031)

```

I am unsure if this is specific to this PR, or has crept in since I last ran in this configuration, but since we already have
this setting in the main server chassis I think it advisable we include in the ui server chassis? cc: @bogdan-sava @lpalashevski

**Comment by bogdan-sava** [comment]
I see!
It starts with the same loader.path as server-chassis and it shouldn't.
I'll add the property, but in the future we should not have the same lib folder.

The dependency is adding a contributor to health check.

**Comment by bogdan-sava** [comment]
Please do not export LOADER_PATH og OMAG to environment. 
Use -Dloader.path in command line.

**Comment by planetf1** [comment]
It's a reflection of what we do currently in the helm chart for coco pharma where the same configuration for LOADER_PATH is used for both. Hence the failure (so good it was spotted :-) 

I agree though in terms of setting the environment. It doesn't make sense to add these libraries into the loader path for the ui chassis. Therefore I'll open up an issue on the egeria-charts repo to update the environment for the pod running the ui chassis so that it has a cleaner environment.

So looks like fine to merge. (with the exposure of it may be a day or two for me to update the chart - will add reference below)

**Comment by bogdan-sava** [comment]
I added the property in application.properties .
However, LOADER_PATH can be overriden by command line.

The order is:
https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config

**Comment by planetf1** [comment]
I've raised the issue for cleanup in the container image
Running another test - if good I propose all good to merge

**Comment by planetf1** [comment]
test passed. recommend merge

**Review by planetf1 — APPROVED** [review]

**Comment by lpalashevski** [comment]
PR looks good.
In regards to actuator healt check behaviour I am suggesting to completely disable this by default for all (not only cassandra)

management.health.defaults.enabled = false

I will check if this is working and create separate PR + document it somewhere in admin docs.

**Pull request merged**